binary-decode.go

```diff
@@ -50,7 +50,7 @@ func (cdc *Codec) decodeReflectBinary(bz []byte, info *TypeInfo, rv reflect.Valu
 		err = errors.New("EOF skipping prefix bytes.")
 		return
 	}
-	if !info.Prefix.EqualBytes(bz) {
+	if !info.Prefix.EqualBytes(bz[:PrefixBytesLen]) {
```

Comment:
I was wondering about this too, I found it in my translation to Unmarshaling from an io.Reader
<img width="677" alt="screen shot 2018-02-23 at 11 23 14 pm" src="https://user-images.githubusercontent.com/4898263/36626957-d9fbbe50-18f0-11e8-95e3-9c87801e2f52.png">

Reply:
I was reading the code and it looked wrong.  heh.

Reply:
High five!